src/arithmetic/algebraic_expression/algebraic_expression.c

```diff
@@ -347,12 +347,15 @@ AlgebraicExpression *AlgebraicExpression_RemoveLeftmostNode
 	AlgebraicExpression **root  // Root from which to remove left most child.
 ) {
 	assert(*root);
+	bool transpose = false;
 	AlgebraicExpression *prev = *root;
 	AlgebraicExpression *current = *root;
 
 	while(current->type == AL_OPERATION) {
+		if(current->operation.op == AL_EXP_TRANSPOSE) transpose = !transpose;
```

Comment:
When calling `AlgebraicExpression_RemoveLeftmostNode` the caller might understand that his expression looks as follows:
`exp = T(*(A,B))` e.g. `Transpose(A*B) ` where `A` is the left most operand and `B` is the rightmost operand and the intension is to remove `A`. in which case the user will want to call `AlgebraicExpression_RemoveLeftmostNode` to remove `A`. but will end up removing `B`

In addition please recall that `Transpose(A + B) = At + Bt ` doesn't perform a switch unlike `Transpose(A * B) = Bt * At`
as such I think this new logic introduce will not perform the expected when dealing with additions.

I would like to better understand the issue this PR resolves and think about other ways to approach it.